Src/Base/AMReX_VisMF.cpp

```diff
@@ -2471,6 +2518,828 @@ VisMF::WriteAsync (const FabArray<FArrayBox>& mf, const std::string& mf_name)
     return af;
 }
 
+void
+VisMF::WriteAsyncMultiFab (const FabArray<FArrayBox>& mf, const std::string& mf_name)
+{
+    BL_PROFILE("VisMF::WriteAsyncMPIAPI()");
+    AMREX_ASSERT(mf_name[mf_name.length() - 1] != '/');
+    static_assert(sizeof(int64_t) == sizeof(Real)*2 or sizeof(int64_t) == sizeof(Real),
+                  "WriteAsync: unsupported Real size");
+
+#ifdef AMREX_USE_MPI
+    int thread_support = -1;
+    MPI_Query_thread(&thread_support);
+    AMREX_ALWAYS_ASSERT(thread_support == MPI_THREAD_MULTIPLE);
+#endif
+
+    const DistributionMapping& dm = mf.DistributionMap();
+
+    int myproc = ParallelDescriptor::MyProc();
+    int nprocs = ParallelDescriptor::NProcs();
+
+    RealDescriptor const& whichRD = []() -> RealDescriptor const& {
+        switch (FArrayBox::getFormat())
+        {
+        case FABio::FAB_NATIVE:
+            return FPC::NativeRealDescriptor();
+        case FABio::FAB_NATIVE_32:
+            return FPC::Native32RealDescriptor();
+        case FABio::FAB_IEEE_32:
+            return FPC::Ieee32NormalRealDescriptor();
+        default:
+            return FPC::NativeRealDescriptor();
+        }
+    }();
+    bool doConvert = whichRD != FPC::NativeRealDescriptor();
+
+    VisMF::Header hdr(mf, VisMF::NFiles, VisMF::Header::Version_v1, false);
+
+    auto myinfo = StaticWriteInfo(myproc);
+    int ifile = std::get<0>(myinfo);   // file #
+    int ispot = std::get<1>(myinfo);   // spot #
+
+    constexpr int sizeof_int64_over_real = sizeof(int64_t) / sizeof(Real);
+    const int n_local_fabs = mf.local_size();
+    const int n_global_fabs = mf.size();
+    const int ncomp = mf.nComp();
+    const long n_fab_reals = 2*ncomp;
+    const long n_fab_int64 = 1;
+    const long n_fab_nums = n_fab_reals*sizeof_int64_over_real + n_fab_int64;
```

Comment:
It should be `/sizeof_int64_ove_real` instead of `*sizeof_int64_over_real`.

Reply:
I believe this is correct as written. sizeof_int64_over_real is the ratio of (int64 size / real size). So, reals * int64s_per_real = int64s. It's also a direct copy from the file async version. Do you agree?

Reply:
No I don't agree.  I sometimes write buggy code.  So you copied the bug.

Reply:
Just thing it this way.  If we switch from double to float, we need less space in `Vector<int64_t> localdata`.  Right?  But currently, `n_fab_nums` is bigger for float than double, because 8/4 is bigger thatn 8/8.

Reply:
I gotcha:
The units are `sizeof_int64_over_real = (bytes/int64)/(bytes/real)` = `reals/int64`.
`number of reals / (reals/int64) = number of int64s`
👍 
I'll adjust and rerun to double check.